app/routes/datasets/dataset.py

```diff
@@ -94,15 +95,24 @@ async def update_dataset(
     dataset: str = Depends(dataset_dependency),
     request: DatasetUpdateIn,
     is_authorized: bool = Depends(is_admin),
+    is_authorized_admin: bool = Depends(is_admin),
 ) -> DatasetResponse:
     """Partially update a dataset.
 
     Only metadata field can be updated. All other fields will be
     ignored.
 
-    Only the dataset owner or a user with `ADMIN` user role can do this operation.
+    Only the dataset owner or a user with `ADMIN` user role can do this operation. Only
+    a user with `ADMIN` role can change the dataset owner.
     """
     input_data: Dict = request.dict(exclude_none=True, by_alias=True)
+
+    if request.owner_id is not None and not is_authorized_admin:
```

Comment:
is there a check here to see if the new `owner_id` has a `MANAGER` role or are we missing that?

Reply:
good caall